pkg/filter/filter.go

```diff
@@ -108,24 +119,34 @@ func JoinResourceTypeMatchers(ms ...ResourceTypeMatcher) ResourceTypeMatcher {
 
 // ResourceRequirement allows specifying a resource requirement by type and/or name
 type ResourceRequirement struct {
+	// Provides the Name of the resource object
 	v1.LocalObjectReference `json:",inline,omitempty"`
+	// Provides the Group, Version, and Resource values (GVR)
 	ResourceTypeRequirement `json:",inline,omitempty"`
+	// Specifies a set of label requirements to be used as filters for matches
+	metav1.LabelSelector `json:",inline,omitempty"`
 }
 
 // Matches returns true if the specified resource name/GVR matches the requirement
-func (r ResourceRequirement) Matches(name string, gvr schema.GroupVersionResource) bool {
+func (r ResourceRequirement) Matches(name string, gvr schema.GroupVersionResource, resourceLabels map[string]string) bool {
 	// If the requirement does not specify a resource name - only check the
 	// ResourceTypeRequirement i.e. GVR match
-	if r.LocalObjectReference.Name == "" {
-		return r.ResourceTypeRequirement.Matches(gvr)
+	matchFound := matches(r.Name, name) && r.ResourceTypeRequirement.Matches(gvr)
```

Comment:
As I think @aaron-kasten mentioned (if not here, maybe elsewhere), this would be cleaner using short retrurns and not using a `matchFound`.  Also I think in case of error we don't want to match, so:
```
if !matches(r.Name, name) || !r.ResourceTypeRequirement.Matches(gvr) {
    return false
}
if len(r.MatchExpressions) == 0 && len(r.MatchLabels) == 0 {
    return true
}
sel, err := metav1.LabelSelectorAsSelector(&metav1.LabelSelector{MatchLabels: r.MatchLabels, MatchExpressions: r.MatchExpressions})
if err != nil {
    // malformed requirement labels
    return false
}
return sel.Matches(labels.Set(resourceLabels))
```

Reply:
I've committed these changes with two minor differences. See comments in code for the explanations.